### DISCUSSION TITLE
RubyEventStore::EventDuplicatedInStream does not provide valuable debugging info

### DISCUSSION THREAD

**Discussion by paneq** [RailsEventStore/rails_event_store#1885] [Ideas]
such as id of duplicated event and stream name.

**Comment by mostlyobvious** [comment]
First of all we do not allow events with duplicate ids, regardless of stream. We allow linking to multiple streams but also disallow linking to same event twice.

For both cases we reuse the same error, which might be fine. But communicating the issue could be different:
- in case of append/publish it doesn't matter which stream you choose, all subsequent appends of that event_id would fail
- in case of link you possibly can succeed by picking different stream name

**Comment by mostlyobvious** [comment]
At least communicating event_id would definitely help
